cli/signals.go

```diff
@@ -54,15 +57,28 @@ func backtrace() {
 }
 
 func fatalSignal(sig syscall.Signal) bool {
-	return sigFatal[sig]
+	_, exists := handledSignalsMap[sig]
+	if !exists {
+		return false
+	}
+
+	return handledSignalsMap[sig]
+}
+
+func nonFatalSignal(sig syscall.Signal) bool {
+	_, exists := handledSignalsMap[sig]
+	if !exists {
+		return false
+	}
+
+	return !handledSignalsMap[sig]
```

Comment:
you already looked  for the signal, I think you could improve the performance doing this

```go
	fatalSignal, exists := handledSignalsMap[sig]
	if !exists {
		return false
	}

	return !faltalSignal
```

Reply:
Good call! Branch updated.